src/Symfony/Component/Yaml/Parser.php

```diff
@@ -800,6 +783,45 @@ private function isStringUnIndentedCollectionItem()
      */
     private function isBlockScalarHeader()
     {
-        return (bool) preg_match('~'.self::BLOCK_SCALAR_HEADER_PATTERN.'$~', $this->currentLine);
+        return (bool) self::preg_match('~'.self::BLOCK_SCALAR_HEADER_PATTERN.'$~', $this->currentLine);
+    }
+
+    /**
+     * A local wrapper for `preg_match` which will throw a ParseException if there
+     * is an internal error in the PCRE engine.
+     *
+     * This avoids us needing to check for "false" every time PCRE is used
+     * in the YAML engine
+     *
+     * @throws ParseException on a PCRE internal error
+     * @see preg_last_error()
+     */
+    static function preg_match($pattern, $subject, &$matches = null, $flags = 0, $offset = 0)
```

Comment:
Should probably be called `pregMatch` if you want to call it like this. Perhaps `match` would be a better name